### PR TITLE
BreadCrumbs for course-detail page.

### PR DESCRIPTION
**Pull request by awais786**
BreadCrumbs implemented for course detail page.
https://openedx.atlassian.net/browse/ECOM-6074

### CODE REVIEW

**Comment by awais786** [comment]
screen shot 2017-01-10 at 5 55 04 pm

**Comment by awais786** [comment]
@tasawernawaz @waheedahmed Please review.

File: course_discovery/apps/publisher/tests/test_views.py [hunk 1]
```diff
@@ -1345,6 +1363,9 @@ def assert_dashboard_response(self, studio_count=0, published_count=0, progress_
         self.assertEqual(len(response.context['published_course_runs']), published_count)
         self.assertEqual(len(response.context['in_progress_course_runs']), progress_count)
         self.assertEqual(len(response.context['preview_course_runs']), preview_count)
+
+        # verify no other page contains any bred crumb related tags.
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/tests/test_views.py]: This is not clear. why any other page can't have breadcrumb ?

**Review by tasawernawaz — APPROVED** [review]
If we want to use this approach by sending breadcrumbs through context, then can you please update existing breadcrumbs on course detail page as well. e,g https://github.com/edx/course-discovery/blob/master/course_discovery/templates/publisher/view_course_form.html#L10-L12

File: course_discovery/apps/publisher/tests/test_views.py (hunk 1 above)
Comment (awais786) [reply course_discovery/apps/publisher/tests/test_views.py]: Because for the time being not every page contains breadcrumbs.

**Comment by awais786** [comment]
@bderusha Please review this PR.

**Comment by awais786** [comment]
@bderusha Please review this PR.

File: course_discovery/apps/publisher/tests/test_views.py [hunk 2]
```diff
@@ -1043,6 +1046,20 @@ def assert_can_edit_permission(self, can_edit=False):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.context['can_edit'], can_edit)
 
+    def _assert_breadcrumbs(self, response, course_run):
+        """ Dry method to assert bread crumbs. """
```
Comment (bderusha) [inline course_discovery/apps/publisher/tests/test_views.py]: `""" Assert breadcrumbs are present in the response. """`

File: course_discovery/apps/publisher/tests/test_views.py (hunk 1 above)
Comment (bderusha) [reply course_discovery/apps/publisher/tests/test_views.py]: I don't think this is necessary

File: course_discovery/templates/publisher/_breadcrumbs.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{% if breadcrumbs %}
+    <ol class="breadcrumb">
+      {% for breadcrumb in breadcrumbs %}
+        <li class="breadcrumb-item {% if forloop.last %}active{% endif %}">
```
Comment (bderusha) [inline course_discovery/templates/publisher/_breadcrumbs.html:4]: `forloop.last` COOL!  I didn't know that this was a thing

**Review by bderusha — APPROVED** [review]
👍 once latest comments addressed

**Pull request merged**